webgl/transcoder/basis_wrappers.cpp

```diff
@@ -39,7 +37,6 @@ struct basis_file
 
     if (!m_transcoder.validate_header(m_file.data(), m_file.size())) {
       m_file.clear();
-      std::cerr << "Invalid Basis header" << std::endl;
```

Comment:
Some sort of error reporting is desirable - at least a bool isValid() method returning (m_magic == MAGIC).

Reply:
Consider `fputs(“Invalid basis header\n”, stderr);` to avoid dragging in all of iostream for one error message

Reply:
Note that error handling can/should be performed by checking the result of `startTranscoding()` call - the extra console output may be valuable for debugging but not as part of the production .wasm binary.

Reply:
... which makes me think that there should be a "return" statement instead of std::cerr :) I'll file a separate PR for this

Reply:
Separately, I think `m_magic` variable and all associated code can be removed based on the fact that every method of transcoder is doing quick header validation which should be ~as strong as the code in this file; not that big of a deal but that seems cleaner. It will also make it so that JS code can call startTranscoding and other functions without hitting assertions in debug builds for malformed files.

Reply:
I agree we can remove the magic. This is old code when we weren't using Embind and had a more C-like api. It was possible to pass a raw pointer to any of the functions. Now that we have a proper `BasisFile` constructor, we could just have the constructor throw a Javascript exception (or have a `createBasisFile` function return `undefined`. That way none of the member functions are accessible unless you first make a valid `BasisFile`. This plus the quick header validation should be sufficient.